### PR TITLE
#454 НСтр формат строкового литерала

### LINKED ISSUE #454: Стандарт 761: НСтр формат строкового литерала
**Название проверки**
НСтр формат строкового литерала

**Английское название проверки**
NStr string literal format

**Правило анализа кода/метаданных**

**Мета-информация (пожалуйста, заполните если знаете):**
- Номер стандарта: 761
- Код проверки: bsl-nstr-string-literal-format
- Тип ошибки: CODE_STYLE
- Критичность: MINOR
- Код ошибки АПК: --

**Параметры проверки**
- checkEmptyInterface 

**Текст ошибки**
- несколько 

**Пример некорректного решения**
-

**Описание, почему так делать нельзя**
-

**Пример корректного решения**

**Дополнительные материалы**
- [Интерфейсные тексты в коде: требования по локализации](https://its.1c.ru/db/v8std#content:761:hdoc)

### PR DESCRIPTION
**Pull request by marmyshev**
Closes #454

### CODE REVIEW

File: bundles/com.e1c.v8codestyle.bsl/markdown/bsl-nstr-string-literal-format.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+```bsl
+Procedure NonCompliant1(Message) Export
+	
+	Message = NStr("en = 'User message'" + Chars.LF);
+	
+EndProcedure
+```
+
+The string literal in first parameter should not be emty.
```
Comment (DoublesunRUS) [inline bundles/com.e1c.v8codestyle.bsl/markdown/bsl-nstr-string-literal-format.md:15]: empty

File: bundles/com.e1c.v8codestyle.bsl/markdown/ru/bsl-nstr-string-literal-format.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# НСтр формат строкового литерала
```
Comment (DoublesunRUS) [inline bundles/com.e1c.v8codestyle.bsl/markdown/ru/bsl-nstr-string-literal-format.md:1]: А где русский вариант?

File: tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+import com._1c.g5.v8.dt.validation.marker.Marker;
+import com.e1c.g5.v8.dt.check.ICheck;
+import com.e1c.g5.v8.dt.testing.check.CheckTestBase;
+import com.e1c.v8codestyle.internal.bsl.BslPlugin;
+
+/**
+ * Abstract test base for loading project and usually update it's single module from {@code test.bsl} resource.
+ * Default project is {@code CommonModule} that allows to load rest resource to common module.
+ *<br><br>
+ * Clients my override the name of the project and path to the module, module id, and the way how to get the module
```
Comment (DoublesunRUS) [inline tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java:43]: may

File: tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+ * (for example manage module, object module, form module and etc).
```
Comment (DoublesunRUS) [inline tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java:44]: manager?

**Review by DoublesunRUS — CHANGES_REQUESTED** [review]

File: tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+import com._1c.g5.v8.dt.bsl.model.Module;
+import com._1c.g5.v8.dt.core.platform.IDtProject;
+import com._1c.g5.v8.dt.metadata.mdclass.CommonModule;
+import com._1c.g5.v8.dt.validation.marker.Marker;
+import com.e1c.g5.v8.dt.check.ICheck;
+import com.e1c.g5.v8.dt.testing.check.CheckTestBase;
+import com.e1c.v8codestyle.internal.bsl.BslPlugin;
+
+/**
+ * Abstract test base for loading project and usually update it's single module from {@code test.bsl} resource.
```
Comment (tretyakevich) [inline tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java:40]: Abstract test base which loads a project and updates its single module from a {@code test.bsl} resource if necessary

File: tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+     */
+    protected String getTestConfigurationName()
+    {
+        return PROJECT_NAME;
+    }
+
+    /**
+     * Gets the module instance from the project.
+     * <br><br>
+     * Clients should override this method if whey need to get module form different path
```
Comment (tretyakevich) [inline tests/com.e1c.v8codestyle.bsl.itests/src/com/e1c/v8codestyle/bsl/check/itests/AbstractSingleModuleTestBase.java:117]: whey -> they

**Review by tretyakevich — APPROVED** [review]

**Pull request merged**